### PR TITLE
Implemented A19 assignment

### PR DESCRIPTION
**Pull request by meenashriTMM**

### CODE REVIEW

File: Parser/Node.cs [hunk 1]
```diff
@@ -48,6 +53,36 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
...
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A Call statement
+public record NCallStmt (Token Name, NExpr[] Args) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// If statement
+public record NIfStmt (NExpr Expr, NStmt[] Stmts) : NStmt {
```
Comment (tarydon) [inline Parser/Node.cs]: This should just be:
`public record NIfStmt (NExpr Expr, NStmt IfPart, NStmt? ElsePart)`
There is one (mandatory) statement for the if part, and an optional statement for the else part. If there are multiple statments in the If, that is basically a single CompoundStmt bracketed by begin..end

File: Parser/Parser.cs [hunk 2]
```diff
@@ -30,17 +30,33 @@ NBlock Block ()
...
 
    // ident-list = IDENT { "," IDENT }
    Token[] IdentList () {
       List<Token> names = new ();
       do { names.Add (Expect (IDENT)); } while (Match (COMMA));
-      return names.ToArray (); 
+      return names.ToArray ();
+   }
+
+   // paramlist = "(" [var-decl { ";" var-decl }] ")"
```
Comment (tarydon) [inline Parser/Parser.cs:53]: Unlike in a var declaration sequence, the variable groups in a parameter list are separated by commas:
`paramlist  =  "(" var-decl { "," var-decl } ")"`

File: Parser/Types/Types.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@ namespace PSI;
 using static NType;
 
 // The data-type at any NExpr node
-public enum NType { Unknown, Int, Real, Bool, String, Char, Error, Void }
+public enum NType { None, Int, Real, Bool, String, Char, Error, Void }
```
Comment (tarydon) [inline Parser/Types/Types.cs]: I think it's better to keep the *Unknown* enumeration as indicating we don't yet know the type. Later, when we determine it to be *None*, we can set that. For procedures, you could use the *Void* type I added at the end.

**Review by tarydon — CHANGES_REQUESTED** [review]
Excellent. Few minor changes required. 
Thanks for fixing the errors in the PAS programs and in the Grammar.

File: Parser/Parser.cs (hunk 2 above)
Comment (meenashriTMM) [reply Parser/Parser.cs:53]: I checked with a online pascal compiler. Code doesn't compile when parameters are separated by comma. It expects semicolon. Found the same in few pascal documentations.

Comment (tarydon) [reply Parser/Parser.cs:53]: OK, let's accept this as the standard definition.

**Pull request closed without merging**